File: Python/Logic-1/cigar_party.py
import unittest

'''

When squirrels get together for a party, they like to have cigars. 
A squirrel party is successful when the number of cigars is between 40 and 60, inclusive. 
Unless it is the weekend, in which case there is no upper bound on the number of cigars.

Return True if the party with the given values is successful, or False otherwise.


cigar_party(30, False) -> False
cigar_party(50, False) -> True
cigar_party(70, True) -> True

'''

def cigar_party(cigars, is_weekend):
  
 if cigars>=40 and is_weekend:
  return True
  
 elif cigars>=40 and cigars<=60 and not is_weekend:
  return True
  
 return False  


class Test_cigar_party(unittest.TestCase):

    def test_1(self):
        self.assertEqual(cigar_party(30, False), False)

    def test_2(self):
        self.assertEqual(cigar_party(50, False), True)

    def test_3(self):
        self.assertEqual(cigar_party(70, True), True)


if __name__ == '__main__':
    unittest.main()       

File: Java/Logic_1/cigarParty.java
/**
 * @author Luis Teixeira
 *
 * @date   28/03/2018
 *
 * @mail   filipe.teixeira.996@gmail.com
 */
package Logic_1;

import static org.junit.Assert.*;

import java.util.HashMap;
import java.util.Map;

import org.junit.Test;

/**

When squirrels get together for a party, they like to have cigars.
A squirrel party is successful when the number of cigars is between
40 and 60, inclusive. Unless it is the weekend, in which case there is
no upper bound on the number of cigars. Return true if the party with the
given values is successful, or false otherwise.


cigarParty(30, false) --> false
cigarParty(50, false) --> true
cigarParty(70, true) --> true
 */
public class cigarParty {

	private static int i = 0;
	
	@Test
	public void Test1()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("cigarParty(30, false) --> false\n");
		assertEquals(false, cigarParty(30, false));
	    i++;
	}
	
	@Test
	public void Test2()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("cigarParty(50, false) --> true\n");
		assertEquals(true, cigarParty(50, true));
	    i++;
	}
	
	@Test
	public void Test3()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("cigarParty(70, true) --> true\n");
		assertEquals(true, cigarParty(70, true));
	    i++;
	}

	
	public boolean cigarParty(int cigars, boolean isWeekend) {
		  
		if (isWeekend == true && cigars>=40) {
			return true;
		}
		
		else {
			if (cigars>=40 && cigars<=60) {
				return true;
			}
			
			else {
				return false;
			}
		}
		
	}

}
